### ISSUE TITLE
Loss implementation seems different than equation (4) in the paper.

### ISSUE THREAD

**Issue by SHMCU** [nv-tlabs/GSCNN#44]
Looks like the loss implementation does not apply argmax to the semantic segmentation logits, right? But in the paper, equation (4) applies argmax before ConvTri and Gradient computation, right?
Which one is the one used to produce the experimental results?

Thank you,
Hai

**Comment by SHMCU** [comment]
The magnitude computation in custom_functions.py, the mag could become 1.0 for all black classes (no object of this class appear), right?
Is this a bug? or intended?

**Comment by tovacinni** [comment]
Addressed in #46
